executor/update.go

```diff
@@ -208,6 +218,34 @@ func (e *UpdateExec) handleErr(colName model.CIStr, rowIdx int, err error) error
 	return err
 }
 
+func (e *UpdateExec) fastComposeNewRow(rowIdx int, oldRow []types.Datum, cols []*table.Column) ([]types.Datum, error) {
+	newRowData := types.CloneRow(oldRow)
+	for _, assign := range e.OrderedList {
+		handleIdx, handleFound := e.tblColPosInfos.FindHandle(assign.Col.Index)
+		if handleFound && e.canNotUpdate(oldRow[handleIdx]) {
```

Comment:
When will it happen?

Reply:
it happens when update table with outer join, outer table with filled NULL can not be updated

this is old logic of composeNewRow - -

Reply:
ok